pkg/controller/controller.go

```diff
@@ -833,3 +837,28 @@ func (c *Controller) DeleteVolumeGroup(contx context.Context, opt *pb.DeleteVolu
 
 	return pb.GenericResponseResult(nil), nil
 }
+func (c *Controller) GetMetrics(context context.Context, opt *pb.GetMetricsOpts) (*pb.GenericResponse, error) {
+	log.Info("in controller get metrics methods")
+
+	var result *[]model.MetricSpec
+	var err error
+
+	if opt.StartTime=="" && opt.EndTime==""{
```

Comment:
Consider to have error checks methods common.

Reply:
https://github.com/opensds/opensds/issues/690